pkg/generate/code/late_initialize.go

```diff
@@ -0,0 +1,321 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package code
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	ackgenconfig "github.com/aws-controllers-k8s/code-generator/pkg/generate/config"
+	"github.com/aws-controllers-k8s/code-generator/pkg/model"
+)
+
+// FindLateInitializedFieldNames outputs the code to create a sorted slice of fieldNames to
+// late initialize. This slice helps with short circuiting the AWSResourceManager.LateInitialize()
+// method if there are no fields to late initialize.
+//
+// Sample Output:
+// var lateInitializeFieldNames = []string{"Name"}
+func FindLateInitializedFieldNames(
+	cfg *ackgenconfig.Config,
+	r *model.CRD,
+	resVarName string,
+	// Number of levels of indentation to use
+	indentLevel int,
+) string {
+	out := ""
+	indent := strings.Repeat("\t", indentLevel)
+	sortedFieldNames, _ := getSortedLateInitFieldsAndConfig(cfg, r)
+	if len(sortedFieldNames) > 0 {
```

Comment:
QQ: would it make sense if we output a `var %s = []string{}` if there are no field names?

Reply:
Yes. Done.